packages/sdk-js/src/GrowthBook.ts

```diff
@@ -632,68 +674,95 @@ export class GrowthBook<
     // Loop through the rules
     if (feature.rules) {
       for (const rule of feature.rules) {
-        // If it's a conditional rule, skip if the condition doesn't pass
-        if (rule.condition && !this._conditionPasses(rule.condition)) {
-          process.env.NODE_ENV !== "production" &&
-            this.log("Skip rule because of condition", {
-              id,
-              rule,
-            });
-          continue;
-        }
-        // If there are filters for who is included (e.g. namespaces)
-        if (rule.filters && this._isFilteredOut(rule.filters)) {
-          process.env.NODE_ENV !== "production" &&
-            this.log("Skip rule because of filters", {
-              id,
-              rule,
-            });
-          continue;
+        // Pre-check if we have an experiment-based sticky bucket, since that supersedes feature rules:
```

Comment:
@jdorn This is a carve out to make sure we are able to bypass attribute based targeting if a sticky bucket was found. A very similar lookup also happens in the experiment's _run() evaluation, but if we're ok with the approach, I can probably de-dupe the lookup. (Lookups are cached in the SDK anyhow so it's not super consequential)

Reply:
made this more abstract and changed how it affects specific rules. Still could use another set of eyes though